lib/ramble/ramble/experiment_set.py

```diff
@@ -82,16 +61,25 @@ def __init__(self, workspace):
         # Set all workspace variables as base variables.
         workspace_vars = workspace.get_workspace_vars()
         workspace_env_vars = workspace.get_workspace_env_vars()
-        self._set_context(self._contexts.base,
+
+        try:
+            self.keywords.check_reserved_keys(workspace_vars)
+        except ramble.keywords.RambleKeywordError as e:
+            raise RambleVariableDefinitionError(
+                f'Workspace variable error: {e}'
+            )
+
+        self._set_context(self._contexts.workspace,
                           workspace.name,
                           workspace_vars,
                           workspace_env_vars)
 
         # Set some base variables from the workspace definition.
-        self.set_base_var(self.mpi_key, workspace.mpi_command)
-        self.set_base_var(self.log_dir_key, workspace.log_dir)
-        self.set_base_var(self.batch_submit_key, workspace.batch_submit)
-        self.set_base_var(self.spec_name_key, '{application_name}')
+        self.set_base_var(self.keywords.mpi_command, workspace.mpi_command)
+        self.set_base_var(self.keywords.log_dir, workspace.log_dir)
+        self.set_base_var(self.keywords.batch_submit, workspace.batch_submit)
+        self.set_base_var(self.keywords.spec_name,
+                          ramble.expander.Expander.expansion_str(self.keywords.application_name))
```

Comment:
Mostly because I find the short-hand hard for a newcomer to understand.

i.e. a fully qualified bit tells them where it's coming from without having to cross reference the imports.

However, I was not consistent in this PR. Even in this file I have a mix of the two, because in one place the fully qualified makes the lines too long (and pretty gross looking) and flake complains about it.

I'm happy to use one or the other though, if that would help readability.

Reply:
I think #1 prio is to be consistent, and #2 is to follow some "nice" style. Let's change this to make them consistent in whatever style you prefer / think is more consistent with the rest of the codebase

Reply:
OK, I pushed an update. Should be ready once the CI finishes.